File: hw/suarez/U3/HWMazeGeneratorASCII/MazeGenerator/src/ec/edu/espe/view/MazeApp.java
package ec.edu.espe.view;

import ec.edu.espe.controller.MazeController;
import ec.edu.espe.model.Maze;

import javax.swing.*;
import java.util.Scanner;

public class MazeApp {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        MazeController controller = new MazeController();

        System.out.println("=== Maze Generator ===");

        System.out.print("Enter maze height (rows): ");
        int rows = scanner.nextInt();

        System.out.print("Enter maze width (columns): ");
        int cols = scanner.nextInt();

        if (rows < 2 || cols < 2) {
            System.out.println("Maze dimensions must be at least 2x2.");
            return;
        }

        controller.createMaze(rows, cols);
        System.out.println("\nGenerated Maze:\n");
        controller.printMazeAscii();

        /* show GUI with maze
        Maze maze = controller.getMaze();
        SwingUtilities.invokeLater(() -> {
            MazeGUI gui = new MazeGUI(maze);
            gui.setVisible(true);
        });*/

        scanner.close();
    }
}

File: hw/fuentes/U3/HW17Maze/MazeAppPY/ec/edu/espe/Maze/view/MazeApp.py
import sys
import os
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from controller.MazeGenerator import MazeGenerator
from view.MazePrinter import print_maze_ascii

def main():
    print("=== Maze Generator ===")
    N = int(input("Enter number of rows (N): "))
    M = int(input("Enter number of columns (M): "))

    generator = MazeGenerator()
    maze = generator.generate_maze(N, M)

    print("\nGenerated Maze:\n")
    print_maze_ascii(maze)

if __name__ == "__main__":
    main()
